src/Microsoft.ComponentDetection.Contracts/TypedComponent/VcpkgComponent.cs

```diff
@@ -71,4 +81,22 @@ public override PackageURL PackageUrl
             }
         }
     }
+
+    private void SetGitRepoProperties()
+    {
+        /* example download locations
+         * "git+https://github.com/leethomason/tinyxml2@9.0.0"
+         * "git+https://github.com/Microsoft/vcpkg#ports/nlohmann-json"
+        */
+        var locationArr = this.DownloadLocation.Split('/');
+        if (!string.IsNullOrEmpty(locationArr[2]))
+        {
+            this.GitRepositoryOwner = locationArr[2];
+        }
+
+        if (!string.IsNullOrEmpty(locationArr[3]))
+        {
+            this.GitRepositoryName = locationArr[3].TakeWhile(ch => char.IsLetterOrDigit(ch)).ToString();
```

Comment:
I think this check may fail for repositories with valid names like `component-detection`